### ISSUE TITLE
TypeScript: Cannot find module 'diary' or its corresponding type declarations

### ISSUE THREAD

**Issue by maxmilton** [maraisr/diary#9]
TypeScript is having trouble finding the package types in the latest version of diary.

I created a minimal repro: https://github.com/maxmilton/repro-diary-types

## CLI output

```
> tsc --noEmit

packages/app-api/src/log.ts:3:23 - error TS2307: Cannot find module 'diary' or its corresponding type declarations.

3 import { diary } from 'diary';
                        ~~~~~~~
```

## Versions

- Node: `v16.2.0`
- Yarn: `v1.22.10`
- TypeScript: `v4.3.2`
- Diary: `v0.1.2`
- OS: `Linux max-l 5.12.7-arch1-1 #1 SMP PREEMPT Wed, 26 May 2021 22:03:57 +0000 x86_64 GNU/Linux`

---

On a side note, I'm liking the new API :1st_place_medal: . Feels quite good for creating custom reporters.

**Comment by maraisr** [comment]
Oh! Thank you so much for the issue. Will have this fixed today. Surprisingly hard to build + emit declarations—obviously still not right 😅

In the mean time you can do this to get unblocked.

```ts
// global.d.ts

declare module 'diary' {
    export * from 'diary/types/index';
}
``` 

> havnt tested this, but _should_ work. If not, there will definitely be a fix out later today.

Glad youre liking it, massive perf and size increase as well 😎

**Comment by maxmilton** [comment]
Creating the module declaration doesn't work either.

Debugging things I found that removing the `typesVersions` field from the diary `package.json` fixes the issue. `typesVersions` isn't actually required anyway since it's for mapping a `.d.ts` file to a specific version range of TypeScript but the types in diary are applicable to any TS version.

**Comment by maxmilton** [comment]
Confirming it is fixed in diary `v0.1.4`, including types in the subpath exports :tada: 

Thank you for getting on it so quickly, appreciate it!
